knowledge-base/pdfviewer-sign-pdfs.md

```diff
@@ -0,0 +1,152 @@
+
+## Description
+
+I want to be able to apply a digital signature to a document in the Telerik UI for Blazor PdfViewer.
+
+## Solution
+
+The PdfViewer does not currently have the capability to manage digitial signatures of the document. However, this is still possible using the Telerik Document Processing Libraries PdfProcessing to programmatically manage the signature and the certificate of the document, while the PdfViewer's sole responsibility is to display the document.
+
+1. Create a custom button in the Blazor PdfViewer to handle the logic for applying the digital signature to the PDF. See [PdfViewer - Custom Toolbar Button](https://docs.telerik.com/blazor-ui/components/pdfviewer/toolbar#custom-tools) for instructions.
+2. Use the PdfProcessing tool to add a digital signature programmatically to the PDF. See [Document Processing - Digital Signature documentation](https://docs.telerik.com/devtools/document-processing/libraries/radpdfprocessing/features/security/digital-signatures) for instructions.
+3. Reload the document into the PdfViewer. See [PdfViewer - Overview](https://docs.telerik.com/blazor-ui/components/pdfviewer/overview) for instructions.
+
+
+## Example
+
+```csharp
+<TelerikPdfViewer Data="@PdfSource">
+    <PdfViewerToolBar>
+        <PdfViewerToolBarOpenTool />
+        <!-- all other buttons here -->
+		<PdfViewerToolBarSeparator />
+		
+		<PdfViewerToolBarCustomTool>
+            <TelerikButton OnClick="@OnSignPdfClick">SIGN</TelerikButton>
+        </PdfViewerToolBarCustomTool>
+    </PdfViewerToolBar>
+</TelerikPdfViewer>
+
+@code {
+    private byte[] PdfSource { get; set; }
+	
+	private async Task OnPdfDownload(PdfViewerDownloadEventArgs args)
+    {
+        args.FileName = "My.pdf";
+    }
+
+        private async Task OnSignPdfClick()
+    {
+        var unsignedDocument = this.PdfSource;
+
+        var signedDocument = SignDocument(this.PdfSource);
+
+        this.PdfSource = signedDocument;
+    }
+
+	private byte[] SignDocument(byte[] unsignedDocumentBytes)
+	{
+        // **** PHASE 1 - Get certificate **** //
+		
+		X509Certificate2 certificate = null;
+
+		// OPTION 1 - BRING YOUR OWN CERTIFICATE FILE
+        // RadPdfProcessing enables you to sign and validate signature fields using standard signature encodings
+        // adbe.x509.rsa_sha1 (PKCS #1)
+        // adbe.pkcs7.sha1 (PKCS #7)
+        // adbe.pkcs7.detached (PKCS #7 Detached)
+
+		//certificate = new System.Security.Cryptography.X509Certificates.X509Certificate2(certificateFilePath, certificateFilePassword);
+
+		// OPTION 2 - USE AN EXISTING CERTIFICATE
+		var x509Store = new X509Store("MY", StoreLocation.CurrentUser);
+		x509Store.Open(OpenFlags.ReadOnly | OpenFlags.OpenExistingOnly);
+		var validCerts = x509Store.Certificates.Find(X509FindType.FindByTimeValid, DateTime.Now, false);
+		
+		// IMPORTANT: This example is just taking the very first valid cert. In a real app, you will selected the cert you want to sign the document with
```

Comment:
```suggestion
		// IMPORTANT: This example is taking the very first valid certificate. In a real app, you need to select the certificate you want to sign the document with.
```

Reply:
Handled, 

Except the "need to" part. The developer can/must select whatever certificate it is from the Store that they use. In most cases the first certificate will be the SSL one, and you don't want that one for the PDF document.